cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -688,6 +688,8 @@ public void deleteArtifact(HttpRequest request, HttpResponder responder,
     try {
       artifactRepository.deleteArtifact(artifactId);
       responder.sendStatus(HttpResponseStatus.OK);
+    } catch (ArtifactNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Artifact " + artifactId + " not found.");
```

Comment:
It seems to be caught in the other methods of the class, like: https://github.com/caskdata/cdap/blob/e804b35df401b2c5a95131c4b7e2d859eaf6d477/cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java#L332

Should I remove it anyway?